### PR TITLE
add user request manager

### PR DESCRIPTION
**Pull request by c0np4nn4**

### CODE REVIEW

File: crates/request-manager/src/handlers/get_storage_value.rs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                            Json(serde_json::json!({
+                                "status": "fetched",
+                                "value": value,
+                            }))
+                            .into_response()
+                        } else {
+                            response_with_status(StatusCode::BAD_REQUEST, "Invalid response format")
+                        }
+                    }
+                    Err(_) => response_with_status(StatusCode::BAD_REQUEST, "Failed to parse JSON"),
```
Comment (ametel01) [inline crates/request-manager/src/handlers/get_storage_value.rs]: consider better error logging eg
```rust
Err(err) => {
    eprintln!("Failed to parse JSON: {:?}", err);
    response_with_status(StatusCode::BAD_REQUEST, "Failed to parse JSON")
},
```

File: crates/request-manager/src/main.rs [hunk 2]
```diff
@@ -62,10 +55,7 @@ async fn main() {
         )
         .with_state(app_state);
 
-    // run our app with hyper, listening globally on port 8000
     let listener = tokio::net::TcpListener::bind("0.0.0.0:8000").await.unwrap();
```
Comment (ametel01) [inline crates/request-manager/src/main.rs]: be more explicit with logging and errors handling eg:

```rust
let listener = match tokio::net::TcpListener::bind("0.0.0.0:8000").await {
        Ok(listener) => {
            info!("Listening on http://{}", listener.local_addr().unwrap());
            listener
        }
        Err(e) => {
            error!("Failed to bind to address: {}", e);
            return;
        }
    };

    if let Err(e) = axum::serve(listener, app).await {
        error!("Server error: {}", e);
    }
```

**Review by ametel01 — CHANGES_REQUESTED** [review]

File: crates/starknet-handler/src/artifacts/FactRegistry.json
Comment (srj31) [inline crates/starknet-handler/src/artifacts/FactRegistry.json:1]: can be deleted, its unnecessary now

File: crates/starknet-handler/src/artifacts/fossil_L1HeaderStore.contract_class.json
Comment (srj31) [inline crates/starknet-handler/src/artifacts/fossil_L1HeaderStore.contract_class.json:1]: can be deleted, its unnecessary

File: crates/starknet-handler/src/interface/mod.rs
Comment (srj31) [inline crates/starknet-handler/src/interface/mod.rs:1]: interface/ is no longer used in the starknet-handler

**Comment by ametel01** [comment]
We are completely missing all the functionalities, the request manager should receive the request from the user exposing an endpoint eg: "get-storage" and call the necessary crates given the sequence diagram. This seems to be only spinning up a server

**Pull request closed without merging**